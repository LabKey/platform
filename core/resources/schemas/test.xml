<?xml version="1.0" encoding="UTF-8"?>
<!-- 

 * Copyright (c) 2004-2017 Fred Hutchinson Cancer Research Center
 *
 * Licensed under the Apache License, Version 2.0 (the "License");
 * you may not use this file except in compliance with the License.
 * You may obtain a copy of the License at
 *
 *     http://www.apache.org/licenses/LICENSE-2.0
 *
 * Unless required by applicable law or agreed to in writing, software
 * distributed under the License is distributed on an "AS IS" BASIS,
 * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
 * See the License for the specific language governing permissions and
 * limitations under the License.

 -->
<tables xsi:schemaLocation="http://labkey.org/data/xml ..\..\..\..\..\schemas\tableInfo.xsd" xmlns="http://labkey.org/data/xml" xmlns:xsi="http://www.w3.org/2001/XMLSchema-instance">
	<table tableName="TestTable" tableDbType="TABLE">
		<columns>
			<column columnName="_ts"/>
			<column columnName="EntityId"/>
			<column columnName="RowId"/>
			<column columnName="CreatedBy"/>
			<column columnName="Created"/>
			<column columnName="Container"/>
			<column columnName="Text"/>
			<column columnName="IntNull"/>
			<column columnName="IntNotNull"/>
			<column columnName="DatetimeNull"/>
			<column columnName="DatetimeNotNull"/>
			<column columnName="RealNull"/>
			<column columnName="BitNull"/>
			<column columnName="BitNotNull"/>
		</columns>
	</table>
    <table tableName="TestTable2" tableDbType="TABLE">
        <columns>
            <column columnName="_ts"/>
            <column columnName="EntityId"/>
            <column columnName="RowId"/>
            <column columnName="CreatedBy"/>
            <column columnName="Created"/>
            <column columnName="Container"/>
            <column columnName="Text"/>
            <column columnName="IntNull"/>
            <column columnName="IntNotNull"/>
            <column columnName="DatetimeNull"/>
            <column columnName="DatetimeNotNull"/>
            <column columnName="RealNull"/>
            <column columnName="BitNull"/>
            <column columnName="BitNotNull"/>
        </columns>
    </table>
    <table tableName="TestTable3" tableDbType="UNKNOWN">
        <description>This is used to test synonyms on SQL Server</description>
        <columns>
            <column columnName="_ts"/>
            <column columnName="EntityId"/>
            <column columnName="RowId"/>
            <column columnName="CreatedBy"/>
            <column columnName="Created"/>
            <column columnName="Container"/>
            <column columnName="Text"/>
            <column columnName="IntNull"/>
            <column columnName="IntNotNull"/>
            <column columnName="DatetimeNull"/>
            <column columnName="DatetimeNotNull"/>
            <column columnName="RealNull"/>
            <column columnName="BitNull"/>
            <column columnName="BitNotNull"/>
        </columns>
    </table>
    <table tableName="Containers2" tableDbType="UNKNOWN">
        <description>This is used to test synonyms on SQL Server</description>
        <columns>
            <column columnName="_ts"/>
            <column columnName="RowId"/>
            <column columnName="EntityId"/>
            <column columnName="CreatedBy"/>
            <column columnName="Created"/>
            <column columnName="Parent"/>
            <column columnName="Name"/>
            <column columnName="SortOrder"/>
            <column columnName="Description"/>
            <column columnName="Type"/>
            <column columnName="Title"/>
            <column columnName="Searchable"/>
        </columns>
    </table>
    <table tableName="ContainerAliases2" tableDbType="UNKNOWN">
        <description>This is used to test synonyms on SQL Server</description>
        <columns>
            <column columnName="Path"/>
            <column columnName="ContainerId"/>
        </columns>
    </table>
    <table tableName="Users2" tableDbType="UNKNOWN">
        <description>This is used to test synonyms on SQL Server</description>
        <columns>
            <column columnName="Email"/>
            <column columnName="_ts"/>
            <column columnName="EntityId"/>
            <column columnName="CreatedBy"/>
            <column columnName="Created"/>
            <column columnName="ModifiedBy"/>
            <column columnName="Modified"/>
            <column columnName="Owner"/>
            <column columnName="UserId"/>
            <column columnName="DisplayName"/>
            <column columnName="FirstName"/>
            <column columnName="LastName"/>
            <column columnName="Phone"/>
            <column columnName="Mobile"/>
            <column columnName="Pager"/>
            <column columnName="IM"/>
            <column columnName="Description"/>
            <column columnName="LastLogin"/>
            <column columnName="Active"/>
            <column columnName="HasPassword"/>
            <column columnName="ExpirationDate"/>
        </columns>
    </table>
</tables>
