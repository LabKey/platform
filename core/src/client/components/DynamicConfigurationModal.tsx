import React, { PureComponent } from 'react';
import { Button, Modal } from 'react-bootstrap';
import ReactBootstrapToggle from 'react-bootstrap-toggle';

import { ActionURL, Ajax } from '@labkey/api';

import { SSOFields } from './SSOFields';
import { DynamicFields, TextInput} from './DynamicFields';
import { AuthConfig, AuthConfigProvider } from "./models";

interface Props {
    authConfig: AuthConfig;
    modalType: AuthConfigProvider;
    configType: string;
    canEdit: boolean;
    title?: string; // Used for title override in DB modal
    updateAuthRowsAfterSave: Function;
    closeModal: Function;
}

interface State {
    errorMessage: string;
    emptyRequiredFields: string[];
    fieldValues: any;
}

export default class DynamicConfigurationModal extends PureComponent<Props, Partial<State>> {
    constructor(props) {
        super(props);

        const {authConfig, modalType} = this.props;
        const fieldValues: any = {};
        this.props.modalType.settingsFields.forEach(field => {
            fieldValues[field.name] = field.name in this.props.authConfig ? this.props.authConfig[field.name] : field.defaultValue;
        });

        if (modalType.sso) {
            fieldValues.auth_header_logo = '';
            fieldValues.auth_login_page_logo = '';
            fieldValues.deletedLogos = [];
            fieldValues.changedFiles = [];
        }

        this.state = {
            errorMessage: '',
            emptyRequiredFields: [],
            fieldValues: { // Contains form values
                enabled: authConfig.enabled,
                description: authConfig.description,
                ...fieldValues // Contains dynamic provider-specific field data generated by server
            }
        };
    }

    saveEditedModal = (): void => {
        const {modalType, authConfig, configType, closeModal, updateAuthRowsAfterSave} = this.props;

        const baseUrl = ActionURL.getBaseURL(true);
        const saveUrl = baseUrl + modalType.saveLink;
        let form = new FormData();

        if (this.areRequiredFieldsEmpty()) {
            return;
        }

        if (authConfig.configuration) {
            form.append('configuration', authConfig.configuration.toString());
        }

        Object.keys(this.state.fieldValues).map(item => {
            form.append(item, this.state.fieldValues[item]);
        });

        Ajax.request({
            url: saveUrl,
            method: 'POST',
            form,
            scope: this,
            failure: function(error) {
                const errorObj = JSON.parse(error.response);
                const errorMessage = errorObj.exception;
                this.setState(() => ({ errorMessage }));
            },
            success: function(result) {
                updateAuthRowsAfterSave(result.response, configType);
                closeModal();
            },
        });
    };

    areRequiredFieldsEmpty = () => {
        // Array of all required fields
        const requiredFields = this.props.modalType.settingsFields.reduce(
            (accum, current) => {
                if (current.required) {
                    accum.push(current.name);
                }
                return accum;
            },
            []
        );

        const emptyRequiredFields = requiredFields.filter(name => (this.state.fieldValues[name] == ''));
        if (emptyRequiredFields.length > 0) {
            this.setState({ emptyRequiredFields });
            return true;
        } else {
            return false;
        }
    };

    onToggle = () => {
        this.setState(state => ({
            ...state,
            fieldValues:{ ...state.fieldValues, enabled: !state.fieldValues.enabled }
        }));
    };

    onChange = (event)  => {
        const { name, value } = event.target;
        this.setState((state) => ({
            ...state,
            fieldValues: {...state.fieldValues, [name]: value}
        }));
    };

    handleDeleteLogo = (value: string) => {
        this.setState(state => ({
            ...state,
            fieldValues:{ ...state.fieldValues, deletedLogos: [ ...state.fieldValues.deletedLogos, value] }
        }));
    };

    checkCheckBox = (name: string) => {
        this.setState((state) => ({
            ...state,
            fieldValues: {...state.fieldValues, [name]: !state.fieldValues[name]}
        }));
    };

    onFileChange = (attachment, logoType: string) => {
        this.setState((state) => ({
            ...state,
            fieldValues: {...state.fieldValues, [logoType]: attachment.first()}
        }));
    };

    onFileRemoval = (name: string) => {
        const changedFiles = this.state.fieldValues.changedFiles;
        if (changedFiles.indexOf(name) === -1) {
            this.setState((state) => ({
                ...state,
                fieldValues: {...state.fieldValues, changedFiles: [...state.fieldValues.changedFiles, name]}
            }));
        }
        this.setState((state) => ({
            ...state,
            fieldValues: {...state.fieldValues, [name]: '' }
        }));
    };

    render() {
        const { modalType, closeModal, canEdit, title, authConfig } = this.props;
        const { emptyRequiredFields, errorMessage, fieldValues } = this.state;
        let queryString;
        if (fieldValues.servers) {
            queryString = {
                server: fieldValues.servers,
                principal: fieldValues.principalTemplate,
                sasl: fieldValues.SASL,
            };
        }

        const isAddNewConfig = title;
        const modalTitle = isAddNewConfig ? 'Add ' + title : 'Configure ' + authConfig.description;
        const finalizeButtonText = isAddNewConfig ? 'Finish' : 'Apply';
        const requiredFieldEmpty = emptyRequiredFields.indexOf("description") !== -1;

        return (
            <Modal show={true} onHide={closeModal}>
                <Modal.Header closeButton>
                    <Modal.Title>
                        {modalTitle}
                    </Modal.Title>
                </Modal.Header>
                <Modal.Body>
                    <div className="modal__top">
                        <span className="bold-text"> Configuration Status </span>
                        <ReactBootstrapToggle
                            onClick={this.onToggle}
                            on="Enabled"
                            off="Disabled"
                            onstyle="primary"
                            active={fieldValues.enabled}
                            className="modal__enable-toggle"
                            disabled={!canEdit}
                        />
                    </div>

                    <div className="bold-text modal__settings-text"> Settings </div>

                    <TextInput
                        onChange={this.onChange}
                        value={fieldValues.description}
                        type="text"
                        canEdit={canEdit}
                        requiredFieldEmpty={requiredFieldEmpty}
                        required={true}
                        name="description"
                        caption="Description"
                    />

                    <DynamicFields
                        fields={modalType.settingsFields}
                        fieldValues={fieldValues}
                        canEdit={canEdit}
                        emptyRequiredFields={emptyRequiredFields}
                        modalType={modalType}
                        onChange={this.onChange}
                        checkCheckBox={this.checkCheckBox}
                        onFileChange={this.onFileChange}
                        onFileRemoval={this.onFileRemoval}
                    />

                    {modalType.sso &&
                        <SSOFields
                            headerLogoUrl={authConfig.headerLogoUrl}
                            loginLogoUrl={authConfig.loginLogoUrl}
                            onFileChange={this.onFileChange}
                            handleDeleteLogo={this.handleDeleteLogo}
                            canEdit={canEdit}
                        />
                    }

                    {modalType.testLink && (
                        <div className="modal__test-button">
                            <Button
                                className="labkey-button"
                                onClick={() =>
                                    window.open(
                                        ActionURL.getBaseURL(true) +
                                        modalType.testLink +
                                        ActionURL.queryString(queryString)
                                    )
                                }
                            >
                                Test
                            </Button>
                        </div>
                    )}

                    <div className="modal__error-message"> {errorMessage} </div>

                    <div className="modal__bottom">
                        <div className="modal__bottom-buttons">
                            <a target="_blank" href={modalType.helpLink} className="modal__help-link" rel="noopener noreferrer">
                                {`More about ${authConfig.provider} authentication`}
                            </a>

                            {canEdit ? (
                                <Button className="labkey-button primary" onClick={() => this.saveEditedModal()}>
                                    {finalizeButtonText}
                                </Button>
                            ) : null}
                        </div>

                        <Button className="labkey-button modal__save-button" onClick={closeModal}>
                            {canEdit ? 'Cancel' : 'Close'}
                        </Button>
                    </div>
                </Modal.Body>
            </Modal>
        );
    }
}
