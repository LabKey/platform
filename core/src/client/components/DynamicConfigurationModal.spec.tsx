import React from 'react';

import { shallow } from 'enzyme';
import EnzymeToJson from 'enzyme-to-json';

import { CAS_MODAL_TYPE, DUO_MODAL_TYPE, CAS_CONFIG, DUO_CONFIG } from '../../../test/data';

import DynamicConfigurationModal from './DynamicConfigurationModal';

describe('<DynamicConfigurationModal/>', () => {
    test('CAS Modal', () => {
        const component =
            <DynamicConfigurationModal
                authConfig={CAS_CONFIG}
                configType="ssoConfigurations"
                modalType={CAS_MODAL_TYPE}
                canEdit={true}
                updateAuthRowsAfterSave={() => {}}
                closeModal={() => {}}
            />;

        const wrapper = shallow(component);
        expect(EnzymeToJson(wrapper)).toMatchSnapshot();
    });

    test('CAS Modal View-only', () => {
        const component =
            <DynamicConfigurationModal
                authConfig={CAS_CONFIG}
                configType="ssoConfigurations"
                modalType={CAS_MODAL_TYPE}
                canEdit={false}
                updateAuthRowsAfterSave={() => {}}
                closeModal={() => {}}
            />;

        const wrapper = shallow(component);
        expect(EnzymeToJson(wrapper)).toMatchSnapshot();
    });

    test('Duo Modal', () => {
        const component = (
            <DynamicConfigurationModal
                authConfig={DUO_CONFIG}
                configType={"secondaryConfigurations"}
                modalType={DUO_MODAL_TYPE}
                canEdit={true}
                updateAuthRowsAfterSave={() => {}}
                closeModal={() => {}}
            />
        );

        const wrapper = shallow(component);
        expect(EnzymeToJson(wrapper)).toMatchSnapshot();
    });

    test('Duo Modal View-only', () => {
        const component = (
            <DynamicConfigurationModal
                authConfig={DUO_CONFIG}
                configType={"secondaryConfigurations"}
                modalType={DUO_MODAL_TYPE}
                canEdit={false}
                updateAuthRowsAfterSave={() => {}}
                closeModal={() => {}}
            />
        );

        const wrapper = shallow(component);
        expect(EnzymeToJson(wrapper)).toMatchSnapshot();
    });
});
