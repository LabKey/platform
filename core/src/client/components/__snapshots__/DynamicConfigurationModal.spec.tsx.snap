// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<DynamicConfigurationModal/> CAS Modal 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      Configure undefined
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <div
      className="modal__top"
    >
      <span
        className="bold-text"
      >
         Configuration Status 
      </span>
      <ReactBootstrapToggle
        active={true}
        className="modal__enable-toggle"
        disabled={false}
        handleClassName=""
        handlestyle="default"
        height=""
        off="Disabled"
        offClassName=""
        offstyle="default"
        on="Enabled"
        onClassName=""
        onClick={[Function]}
        onstyle="primary"
        recalculateOnResize={false}
        size="normal"
        style={Object {}}
        width=""
      />
    </div>
    <div
      className="bold-text modal__settings-text"
    >
       Settings 
    </div>
    <TextInput
      canEdit={true}
      caption="Description"
      name="description"
      onChange={[Function]}
      required={true}
      requiredFieldEmpty={false}
      type="text"
    />
    <DynamicFields
      SASL=""
      auth_header_logo=""
      auth_login_page_logo=""
      autoRedirect={false}
      canEdit={true}
      changedFiles={Array []}
      checkCheckBox={[Function]}
      deletedLogos={Array []}
      emptyRequiredFields={Array []}
      errorMessage=""
      fields={
        Array [
          Object {
            "caption": "CAS Server URL",
            "defaultValue": "",
            "description": "Enter a valid HTTPS URL to your CAS server. The URL should start with https:// and end with /cas, for example: https://test.org/cas.",
            "name": "serverUrl",
            "required": true,
            "type": "input",
          },
          Object {
            "caption": "Default to CAS Login",
            "defaultValue": false,
            "description": "Redirects the login page directly to the CAS login instead of requiring the user click the CAS option.",
            "name": "autoRedirect",
            "required": false,
            "type": "checkbox",
          },
        ]
      }
      modalType={
        Object {
          "description": "Central Authentication Service (CAS)",
          "helpLink": "https://www.labkey.org/Documentation/19.3/wiki-page.view?name=configureCas",
          "saveLink": "/labkey/casclient-casSaveConfiguration.view?",
          "settingsFields": Array [
            Object {
              "caption": "CAS Server URL",
              "defaultValue": "",
              "description": "Enter a valid HTTPS URL to your CAS server. The URL should start with https:// and end with /cas, for example: https://test.org/cas.",
              "name": "serverUrl",
              "required": true,
              "type": "input",
            },
            Object {
              "caption": "Default to CAS Login",
              "defaultValue": false,
              "description": "Redirects the login page directly to the CAS login instead of requiring the user click the CAS option.",
              "name": "autoRedirect",
              "required": false,
              "type": "checkbox",
            },
          ],
          "sso": true,
        }
      }
      onChange={[Function]}
      onFileChange={[Function]}
      onFileRemoval={[Function]}
      principalTemplate=""
      search={false}
      serverUrl=""
      servers=""
    />
    <SSOFields
      canEdit={true}
      handleDeleteLogo={[Function]}
      onFileChange={[Function]}
    />
    <div
      className="modal__error-message"
    >
       
       
    </div>
    <div
      className="modal__bottom"
    >
      <div
        className="modal__bottom-buttons"
      >
        <a
          className="modal__help-link"
          href="https://www.labkey.org/Documentation/19.3/wiki-page.view?name=configureCas"
          target="_blank"
        >
          More about CAS authentication
        </a>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="labkey-button primary"
          disabled={false}
          onClick={[Function]}
        >
          Apply
        </Button>
      </div>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        className="labkey-button modal__save-button"
        disabled={false}
        onClick={[Function]}
      >
        Cancel
      </Button>
    </div>
  </ModalBody>
</Modal>
`;

exports[`<DynamicConfigurationModal/> CAS Modal View-only 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      Configure undefined
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <div
      className="modal__top"
    >
      <span
        className="bold-text"
      >
         Configuration Status 
      </span>
      <ReactBootstrapToggle
        active={true}
        className="modal__enable-toggle"
        disabled={true}
        handleClassName=""
        handlestyle="default"
        height=""
        off="Disabled"
        offClassName=""
        offstyle="default"
        on="Enabled"
        onClassName=""
        onClick={[Function]}
        onstyle="primary"
        recalculateOnResize={false}
        size="normal"
        style={Object {}}
        width=""
      />
    </div>
    <div
      className="bold-text modal__settings-text"
    >
       Settings 
    </div>
    <TextInput
      canEdit={false}
      caption="Description"
      name="description"
      onChange={[Function]}
      required={true}
      requiredFieldEmpty={false}
      type="text"
    />
    <DynamicFields
      SASL=""
      auth_header_logo=""
      auth_login_page_logo=""
      autoRedirect={false}
      canEdit={false}
      changedFiles={Array []}
      checkCheckBox={[Function]}
      deletedLogos={Array []}
      emptyRequiredFields={Array []}
      errorMessage=""
      fields={
        Array [
          Object {
            "caption": "CAS Server URL",
            "defaultValue": "",
            "description": "Enter a valid HTTPS URL to your CAS server. The URL should start with https:// and end with /cas, for example: https://test.org/cas.",
            "name": "serverUrl",
            "required": true,
            "type": "input",
          },
          Object {
            "caption": "Default to CAS Login",
            "defaultValue": false,
            "description": "Redirects the login page directly to the CAS login instead of requiring the user click the CAS option.",
            "name": "autoRedirect",
            "required": false,
            "type": "checkbox",
          },
        ]
      }
      modalType={
        Object {
          "description": "Central Authentication Service (CAS)",
          "helpLink": "https://www.labkey.org/Documentation/19.3/wiki-page.view?name=configureCas",
          "saveLink": "/labkey/casclient-casSaveConfiguration.view?",
          "settingsFields": Array [
            Object {
              "caption": "CAS Server URL",
              "defaultValue": "",
              "description": "Enter a valid HTTPS URL to your CAS server. The URL should start with https:// and end with /cas, for example: https://test.org/cas.",
              "name": "serverUrl",
              "required": true,
              "type": "input",
            },
            Object {
              "caption": "Default to CAS Login",
              "defaultValue": false,
              "description": "Redirects the login page directly to the CAS login instead of requiring the user click the CAS option.",
              "name": "autoRedirect",
              "required": false,
              "type": "checkbox",
            },
          ],
          "sso": true,
        }
      }
      onChange={[Function]}
      onFileChange={[Function]}
      onFileRemoval={[Function]}
      principalTemplate=""
      search={false}
      serverUrl=""
      servers=""
    />
    <SSOFields
      canEdit={false}
      handleDeleteLogo={[Function]}
      onFileChange={[Function]}
    />
    <div
      className="modal__error-message"
    >
       
       
    </div>
    <div
      className="modal__bottom"
    >
      <div
        className="modal__bottom-buttons"
      >
        <a
          className="modal__help-link"
          href="https://www.labkey.org/Documentation/19.3/wiki-page.view?name=configureCas"
          target="_blank"
        >
          More about CAS authentication
        </a>
      </div>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        className="labkey-button modal__save-button"
        disabled={false}
        onClick={[Function]}
      >
        Close
      </Button>
    </div>
  </ModalBody>
</Modal>
`;

exports[`<DynamicConfigurationModal/> Duo Modal 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      Configure undefined
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <div
      className="modal__top"
    >
      <span
        className="bold-text"
      >
         Configuration Status 
      </span>
      <ReactBootstrapToggle
        active={true}
        className="modal__enable-toggle"
        disabled={false}
        handleClassName=""
        handlestyle="default"
        height=""
        off="Disabled"
        offClassName=""
        offstyle="default"
        on="Enabled"
        onClassName=""
        onClick={[Function]}
        onstyle="primary"
        recalculateOnResize={false}
        size="normal"
        style={Object {}}
        width=""
      />
    </div>
    <div
      className="bold-text modal__settings-text"
    >
       Settings 
    </div>
    <TextInput
      canEdit={true}
      caption="Description"
      name="description"
      onChange={[Function]}
      required={true}
      requiredFieldEmpty={false}
      type="text"
    />
    <DynamicFields
      SASL=""
      apiHostname=""
      auth_header_logo=""
      auth_login_page_logo=""
      canEdit={true}
      changedFiles={Array []}
      checkCheckBox={[Function]}
      deletedLogos={Array []}
      emptyRequiredFields={Array []}
      errorMessage=""
      fields={
        Array [
          Object {
            "caption": "Integration Key",
            "defaultValue": "",
            "description": "Your Labkey Admin with a Duo administrative account should have generated this key.",
            "name": "integrationKey",
            "required": true,
            "type": "input",
          },
          Object {
            "caption": "Secret Key",
            "defaultValue": "",
            "description": "Your Labkey Admin with a Duo administrative account should have generated this key.",
            "name": "secretKey",
            "required": true,
            "type": "input",
          },
          Object {
            "caption": "API Hostname",
            "defaultValue": "",
            "description": "Your Labkey Admin with a Duo administrative account should have the hostname along with Integration and Secret Key.",
            "name": "apiHostname",
            "required": true,
            "type": "input",
          },
          Object {
            "caption": "User Identifier",
            "defaultValue": "UserID",
            "description": "Choose the way LabKey will match with Duo accounts.",
            "name": "userIdentifier",
            "options": Object {
              "FullEmailAddress": "Full Email Address",
              "UserID": "User ID",
              "UserName": "User Name",
            },
            "required": true,
            "type": "options",
          },
        ]
      }
      integrationKey=""
      modalType={
        Object {
          "description": "Require two-factor authentication via Duo",
          "helpLink": "https://www.labkey.org/Documentation/19.3/wiki-page.view?name=configureDuoTwoFactor",
          "saveLink": "/labkey/duo-duoSaveConfiguration.view?",
          "settingsFields": Array [
            Object {
              "caption": "Integration Key",
              "defaultValue": "",
              "description": "Your Labkey Admin with a Duo administrative account should have generated this key.",
              "name": "integrationKey",
              "required": true,
              "type": "input",
            },
            Object {
              "caption": "Secret Key",
              "defaultValue": "",
              "description": "Your Labkey Admin with a Duo administrative account should have generated this key.",
              "name": "secretKey",
              "required": true,
              "type": "input",
            },
            Object {
              "caption": "API Hostname",
              "defaultValue": "",
              "description": "Your Labkey Admin with a Duo administrative account should have the hostname along with Integration and Secret Key.",
              "name": "apiHostname",
              "required": true,
              "type": "input",
            },
            Object {
              "caption": "User Identifier",
              "defaultValue": "UserID",
              "description": "Choose the way LabKey will match with Duo accounts.",
              "name": "userIdentifier",
              "options": Object {
                "FullEmailAddress": "Full Email Address",
                "UserID": "User ID",
                "UserName": "User Name",
              },
              "required": true,
              "type": "options",
            },
          ],
        }
      }
      onChange={[Function]}
      onFileChange={[Function]}
      onFileRemoval={[Function]}
      principalTemplate=""
      search={false}
      secretKey=""
      servers=""
      userIdentifier="UserID"
    />
    <div
      className="modal__error-message"
    >
       
       
    </div>
    <div
      className="modal__bottom"
    >
      <div
        className="modal__bottom-buttons"
      >
        <a
          className="modal__help-link"
          href="https://www.labkey.org/Documentation/19.3/wiki-page.view?name=configureDuoTwoFactor"
          target="_blank"
        >
          More about Duo 2 Factor authentication
        </a>
        <Button
          active={false}
          block={false}
          bsClass="btn"
          bsStyle="default"
          className="labkey-button primary"
          disabled={false}
          onClick={[Function]}
        >
          Apply
        </Button>
      </div>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        className="labkey-button modal__save-button"
        disabled={false}
        onClick={[Function]}
      >
        Cancel
      </Button>
    </div>
  </ModalBody>
</Modal>
`;

exports[`<DynamicConfigurationModal/> Duo Modal View-only 1`] = `
<Modal
  animation={true}
  autoFocus={true}
  backdrop={true}
  bsClass="modal"
  dialogComponentClass={[Function]}
  enforceFocus={true}
  keyboard={true}
  manager={
    ModalManager {
      "add": [Function],
      "containers": Array [],
      "data": Array [],
      "handleContainerOverflow": true,
      "hideSiblingNodes": true,
      "isTopModal": [Function],
      "modals": Array [],
      "remove": [Function],
    }
  }
  onHide={[Function]}
  renderBackdrop={[Function]}
  restoreFocus={true}
  show={true}
>
  <ModalHeader
    bsClass="modal-header"
    closeButton={true}
    closeLabel="Close"
  >
    <ModalTitle
      bsClass="modal-title"
      componentClass="h4"
    >
      Configure undefined
    </ModalTitle>
  </ModalHeader>
  <ModalBody
    bsClass="modal-body"
    componentClass="div"
  >
    <div
      className="modal__top"
    >
      <span
        className="bold-text"
      >
         Configuration Status 
      </span>
      <ReactBootstrapToggle
        active={true}
        className="modal__enable-toggle"
        disabled={true}
        handleClassName=""
        handlestyle="default"
        height=""
        off="Disabled"
        offClassName=""
        offstyle="default"
        on="Enabled"
        onClassName=""
        onClick={[Function]}
        onstyle="primary"
        recalculateOnResize={false}
        size="normal"
        style={Object {}}
        width=""
      />
    </div>
    <div
      className="bold-text modal__settings-text"
    >
       Settings 
    </div>
    <TextInput
      canEdit={false}
      caption="Description"
      name="description"
      onChange={[Function]}
      required={true}
      requiredFieldEmpty={false}
      type="text"
    />
    <DynamicFields
      SASL=""
      apiHostname=""
      auth_header_logo=""
      auth_login_page_logo=""
      canEdit={false}
      changedFiles={Array []}
      checkCheckBox={[Function]}
      deletedLogos={Array []}
      emptyRequiredFields={Array []}
      errorMessage=""
      fields={
        Array [
          Object {
            "caption": "Integration Key",
            "defaultValue": "",
            "description": "Your Labkey Admin with a Duo administrative account should have generated this key.",
            "name": "integrationKey",
            "required": true,
            "type": "input",
          },
          Object {
            "caption": "Secret Key",
            "defaultValue": "",
            "description": "Your Labkey Admin with a Duo administrative account should have generated this key.",
            "name": "secretKey",
            "required": true,
            "type": "input",
          },
          Object {
            "caption": "API Hostname",
            "defaultValue": "",
            "description": "Your Labkey Admin with a Duo administrative account should have the hostname along with Integration and Secret Key.",
            "name": "apiHostname",
            "required": true,
            "type": "input",
          },
          Object {
            "caption": "User Identifier",
            "defaultValue": "UserID",
            "description": "Choose the way LabKey will match with Duo accounts.",
            "name": "userIdentifier",
            "options": Object {
              "FullEmailAddress": "Full Email Address",
              "UserID": "User ID",
              "UserName": "User Name",
            },
            "required": true,
            "type": "options",
          },
        ]
      }
      integrationKey=""
      modalType={
        Object {
          "description": "Require two-factor authentication via Duo",
          "helpLink": "https://www.labkey.org/Documentation/19.3/wiki-page.view?name=configureDuoTwoFactor",
          "saveLink": "/labkey/duo-duoSaveConfiguration.view?",
          "settingsFields": Array [
            Object {
              "caption": "Integration Key",
              "defaultValue": "",
              "description": "Your Labkey Admin with a Duo administrative account should have generated this key.",
              "name": "integrationKey",
              "required": true,
              "type": "input",
            },
            Object {
              "caption": "Secret Key",
              "defaultValue": "",
              "description": "Your Labkey Admin with a Duo administrative account should have generated this key.",
              "name": "secretKey",
              "required": true,
              "type": "input",
            },
            Object {
              "caption": "API Hostname",
              "defaultValue": "",
              "description": "Your Labkey Admin with a Duo administrative account should have the hostname along with Integration and Secret Key.",
              "name": "apiHostname",
              "required": true,
              "type": "input",
            },
            Object {
              "caption": "User Identifier",
              "defaultValue": "UserID",
              "description": "Choose the way LabKey will match with Duo accounts.",
              "name": "userIdentifier",
              "options": Object {
                "FullEmailAddress": "Full Email Address",
                "UserID": "User ID",
                "UserName": "User Name",
              },
              "required": true,
              "type": "options",
            },
          ],
        }
      }
      onChange={[Function]}
      onFileChange={[Function]}
      onFileRemoval={[Function]}
      principalTemplate=""
      search={false}
      secretKey=""
      servers=""
      userIdentifier="UserID"
    />
    <div
      className="modal__error-message"
    >
       
       
    </div>
    <div
      className="modal__bottom"
    >
      <div
        className="modal__bottom-buttons"
      >
        <a
          className="modal__help-link"
          href="https://www.labkey.org/Documentation/19.3/wiki-page.view?name=configureDuoTwoFactor"
          target="_blank"
        >
          More about Duo 2 Factor authentication
        </a>
      </div>
      <Button
        active={false}
        block={false}
        bsClass="btn"
        bsStyle="default"
        className="labkey-button modal__save-button"
        disabled={false}
        onClick={[Function]}
      >
        Close
      </Button>
    </div>
  </ModalBody>
</Modal>
`;
